### DISCUSSION TITLE
Don't know why this error happens

### DISCUSSION THREAD

**Discussion by jimmyfu0729** [isaac-sim/IsaacLab#81] [Q&A] [answered]
I want to import the pepper robot into the orbit environment, but when I execute my code, the error below appears.

Traceback (most recent call last):
  File "Asset/play_pepper.py", line 125, in <module>
    main()
  File "Asset/play_pepper.py", line 77, in main
    robot.spawn("/pepper", translation=(0.0, 0.0, 1.0))
  File "/home/cjf/Desktop/Isaac/orbit/Orbit/source/extensions/omni.isaac.orbit/omni/isaac/orbit/robots/single_arm/single_arm.py", line 72, in spawn
    super().spawn(prim_path, translation, orientation)
  File "/home/cjf/Desktop/Isaac/orbit/Orbit/source/extensions/omni.isaac.orbit/omni/isaac/orbit/robots/robot_base.py", line 139, in spawn
    kit_utils.set_articulation_properties(prim_path, **self.cfg.articulation_props.to_dict())
  File "/home/cjf/Desktop/Isaac/orbit/Orbit/source/extensions/omni.isaac.orbit/omni/isaac/orbit/utils/kit.py", line 274, in set_articulation_properties
    raise ValueError(f"No articulation schema present for prim '{prim_path}'.")
ValueError: No articulation schema present for prim '/pepper'.

**Comment by jimmyfu0729** [comment]
Or is there anything I need to take care when I try to add a robot not in the library?

**Reply by Mayankm96** [reply, marked as answer]
Does the USD file have the default prim as the articulation?

**Reply by jimmyfu0729** [reply]
I just start to use Isaac Sim, I imported the urdf model to usd but I merged the fixed links. I dont know if this will affect the loading.  I think the articulation is automatically transfered to usd from urdf? All the joints have the same name as the default. But since there is no configuration class of humanoid robot, so I dont know if using this class is correct.

**Reply by jimmyfu0729** [reply]
> Does the USD file have the default prim as the articulation?

This is my configuration file, as there is no class of humanoid so I used MobileManipulatorCfg, but there should be two end effectors .

from omni.isaac.orbit.actuators.group import ActuatorGroupCfg
from omni.isaac.orbit.actuators.group.actuator_group_cfg import ActuatorControlCfg
from omni.isaac.orbit.actuators.model import ImplicitActuatorCfg
from omni.isaac.orbit.robots.single_arm import SingleArmManipulatorCfg
from omni.isaac.orbit.robots.mobile_manipulator import MobileManipulator
from omni.isaac.orbit.robots.mobile_manipulator import MobileManipulatorCfg

PEPPER_USD_PATH="/home/cjf/Desktop/Isaac/orbit/Orbit/Asset/pepper_description/urdf/pepper/pepper.usd"

PEPPER_CFG = MobileManipulatorCfg(
    meta_info=MobileManipulatorCfg.MetaInfoCfg(
        usd_path=PEPPER_USD_PATH,
	base_num_dof=6,
	arm_num_dof=33,
	tool_num_dof=10,
	tool_sites_names=["LFinger_13",
			  "LFinger_23",
			  "LFinger_33",
			  "LFinger_43",
			  "LThumb_2",
			  "RFinger_13",
			  "RFinger_23",
			  "RFinger_33",
			  "RFinger_43",
			  "RThumb_2",],

    ),
    init_state=MobileManipulatorCfg.InitialStateCfg(
        dof_pos={
            # pepper base
	   "BWheel_Yaw": 0.0,
	   "BWheel_Pitch": 0.0,
	   "FLWheel_Yaw": 0.0,
	   "FLWheel_Pitch": 0.0,
	   "FRWheel_Yaw": 0.0,
	   "FRWheel_Pitch": 0.0,
            # pepper body
	   "Knee_Pitch": 0.0,
	   "Hip_Pitch": 0.0,
	   "Hip_Roll": 0.0,
           "Head_Pitch": 0.0,
           "Head_Yaw": 0.0,
            # pepper left arm
           "LShoulderPitch": 0.0,
	   "LShoulderRoll": 0.0,
	   "LElbowYaw": 0.0,
	   "LElbowRoll": 0.0,
           "LWristYaw": 0.0,
           "LWristRoll": 0.0,
	    # pepper right arm
           "RShoulderPitch": 0.0,
	   "RShoulderRoll": 0.0,
	   "RElbowYaw": 0.0,
	   "RElbowRoll": 0.0,
           "RWristYaw": 0.0,
           "RWristRoll": 0.0,
	    # pepper left hand
	   "LFinger_11": 0.0,
	   "LFinger_21": 0.0,
           "LFinger_31": 0.0,
           "LFinger_41": 0.0,
	   "LHand": 0.0,
	   "LFinger_12": 0.0,
	   "LFinger_22": 0.0,
	   "LFinger_32": 0.0,
	   "LFinger_42": 0.0,
	   "LFinger_13": 0.0,
	   "LFinger_23": 0.0,
	   "LFinger_33": 0.0,
	   "LFinger_43": 0.0,
	   "LThumb_1": 0.0,
	   "LThumb_2": 0.0,
	    # pepper right hand
	   "RFinger_11": 0.0,
	   "RFinger_21": 0.0,
           "RFinger_31": 0.0,
           "RFinger_41": 0.0,
	   "RHand": 0.0,
	   "RFinger_12": 0.0,
	   "RFinger_22": 0.0,
	   "RFinger_32": 0.0,
	   "RFinger_42": 0.0,
	   "RFinger_13": 0.0,
	   "RFinger_23": 0.0,
	   "RFinger_33": 0.0,
	   "RFinger_43": 0.0,
	   "RThumb_1": 0.0,
	   "RThumb_2": 0.0,
        },
        dof_vel={".*": 0.0},
    ),
    ee_info=MobileManipulatorCfg.EndEffectorFrameCfg(
        body_name="i_wrist", pos_offset=(0.0, 0.0, 0.0), rot_offset=(1.0, 0.0, 0.0, 0.0)
    ),
    rigid_props=SingleArmManipulatorCfg.RigidBodyPropertiesCfg(
        max_depenetration_velocity=5.0,
    ),
    collision_props=SingleArmManipulatorCfg.CollisionPropertiesCfg(
        contact_offset=0.005,
        rest_offset=0.0,
    ),
    articulation_props=SingleArmManipulatorCfg.ArticulationRootPropertiesCfg(
        enable_self_collisions=True,
    ),
    actuator_groups={
        "pepper_base": ActuatorGroupCfg(
            dof_names=["BWheel_Yaw",
		       "BWheel_Pitch",
		       "FLWheel_Yaw",
		       "FLWheel_Pitch",
		       "FRWheel_Yaw",
		       "FRWheel_Pitch"
		      ],
            model_cfg=ImplicitActuatorCfg(velocity_limit=100.0, torque_limit=87.0),
            control_cfg=ActuatorControlCfg(
                command_types=["v_abs"],
                stiffness={".*": 800.0},
                damping={".*": 40.0},
                dof_pos_offset={
                    "BWheel_Yaw": 0.0,
                    "BWheel_Pitch": 0.0,
                    "FLWheel_Yaw": 0.0,
		    "FLWheel_Pitch": 0.0,
		    "FRWheel_Yaw": 0.0,
		    "FRWheel_Pitch": 0.0
                },
            ),
        ),
        "pepper_body": ActuatorGroupCfg(
            dof_names=["Knee_Pitch"
		       "Hip_Pitch",
		       "Hip_Roll",
                       "Head_Pitch",
                       "Head_Yaw"],
            model_cfg=ImplicitActuatorCfg(velocity_limit=100.0, torque_limit=54.0),
            control_cfg=ActuatorControlCfg(
                command_types=["p_abs"],
                stiffness={".*": 800.0},
                damping={".*": 40.0},
                dof_pos_offset={"Knee_Pitch":0.0,
				"HipRoll": 0.0, 
				"HipPitch": 0.0, 
				"HeadPitch": 0.0, 
				"HeadYaw":0.0
		},
            ),
        ),
        "pepper_L_arm": ActuatorGroupCfg(
            dof_names=["LShoulderPitch",
		       "LShoulderRoll",
	               "LElbowYaw",
		       "LElbowRoll",
		       "LWristYaw",
	               "LWristRoll"],
            model_cfg=ImplicitActuatorCfg(velocity_limit=50.0, torque_limit=25.0),
            control_cfg=ActuatorControlCfg(
                command_types=["p_abs"],
                stiffness={".*": 800.0},
                damping={".*": 40.0},
                dof_pos_offset={"LShoulderPitch": 0.0, 
				"LShoulderRoll": 0.0,
				"LElbowYaw": 0.0,
				"LElbowRoll": 0.0,
				"LWristYaw": 0.0,
				"LWristRoll": 0.0,
		},
            ),
        ),
	"pepper_R_arm": ActuatorGroupCfg(
            dof_names=["RShoulderPitch",
		       "RShoulderRoll",
	               "RElbowYaw",
		       "RElbowRoll",
		       "RWristYaw",
	               "RWristRoll"],
            model_cfg=ImplicitActuatorCfg(velocity_limit=50.0, torque_limit=25.0),
            control_cfg=ActuatorControlCfg(
                command_types=["p_abs"],
                stiffness={".*": 800.0},
                damping={".*": 40.0},
                dof_pos_offset={"RShoulderPitch": 0.0, 
				"RShoulderRoll": 0.0,
				"RElbowYaw": 0.0,
				"RElbowRoll": 0.0,
				"RWristYaw": 0.0,
				"RWristRoll": 0.0,
		},
            ),
        ),
        "pepper_L_hand": ActuatorGroupCfg(
            dof_names=["LFinger_11",
		       "LFinger_21",
	               "LFinger_31",
		       "LFinger_41",
		       "LHand",
		       "LFinger_12"
		       "LFinger_22",
	               "LFinger_32",
		       "LFinger_42",
		       "LFinger_13"
		       "LFinger_23",
	               "LFinger_33",
		       "LFinger_43",
	               "LThumb_1",
	               "LThumb_2",
		      ],
            model_cfg=ImplicitActuatorCfg(velocity_limit=50.0, torque_limit=3.0),
            control_cfg=ActuatorControlCfg(
                command_types=["p_abs"],
                stiffness={".*": 800.0},
                damping={".*": 40.0},
                dof_pos_offset={"LFinger11": 0.0, 
				"LFinger12": 0.0,
				"LFinger13": 0.0,
				"LFinger21": 0.0,
				"LFinger22": 0.0,
				"LFinger23": 0.0,
				"LFinger31": 0.0,
				"LFinger32": 0.0,
				"LFinger33": 0.0,
				"LFinger41": 0.0,
				"LFinger42": 0.0,
				"LFinger43": 0.0,
				"LThumb1": 0.0,
				"LThumb2": 0.0,
				"LHand": 0.0,
		},
            ),
        ),

        "pepper_L_hand": ActuatorGroupCfg(
            dof_names=["RFinger_11",
		       "RFinger_21",
	               "RFinger_31",
		       "RFinger_41",
		       "RHand",
		       "RFinger_12",
		       "RFinger_22",
	               "RFinger_32",
		       "RFinger_42",
		       "RFinger_13",
		       "RFinger_23",
	               "RFinger_33",
		       "RFinger_43",
	               "RThumb_1",
	               "RThumb_2",
		      ],
            model_cfg=ImplicitActuatorCfg(velocity_limit=50.0, torque_limit=3.0),
            control_cfg=ActuatorControlCfg(
                command_types=["p_abs"],
                stiffness={".*": 800.0},
                damping={".*": 40.0},
                dof_pos_offset={"RFinger11": 0.0, 
				"RFinger12": 0.0,
				"RFinger13": 0.0,
				"RFinger21": 0.0,
				"RFinger22": 0.0,
				"RFinger23": 0.0,
				"RFinger31": 0.0,
				"RFinger32": 0.0,
				"RFinger33": 0.0,
				"RFinger41": 0.0,
				"RFinger42": 0.0,
				"RFinger43": 0.0,
				"RThumb1": 0.0,
				"RThumb2": 0.0,
				"RHand": 0.0,
		},
            ),
        ),
    },
)

**Reply by BurgerAndreas** [reply]
Can you explain how you fixed the error? What does one add the `default prim as the articulation`? :)

**Reply by Mayankm96** [reply]
Maybe this helps:

https://www.youtube.com/watch?v=UyqwFWFpQfY&ab_channel=OrdinaryAlan
https://docs.omniverse.nvidia.com/dev-guide/latest/programmer_ref/usd/stage/set-default-prim.html
